lib/types.ts

```diff
@@ -0,0 +1,4 @@
+export interface GenericCallback<T> {
+    (err: null, data: T): void;
+    (err: Error): void;
+};
```

Comment:
We have a very similar type defined in `lib/storage/metadata/mongoclient/MongoClientInterface.ts`, that now makes sense to move here:

```ts
interface ArsenalCallback<T> {
    (err: null, result: T): void;
    (err: void): void;
    (err: ArsenalError): void;
};
```

It also has the benefit of enforcing that we either have an error and no data, or data but no error.